PCAP and VDE network transports require linking with userspace
libraries. The current build system has no means of passing these
as arguments.

This patch adds a script to expand the library list for linking
for these transports as well as any future driver that needs to
rely on additional libraries on the userspace side.

Signed-off-by: Anton Ivanov <anton.ivanov@cambridgegreys.com>
---
 arch/um/scripts/extra-libs.sh | 10 ++++++++++
 scripts/link-vmlinux.sh       |  4 +++-
 2 files changed, 13 insertions(+), 1 deletion(-)
 create mode 100644 arch/um/scripts/extra-libs.sh

Index: user-mode-linux/arch/um/scripts/extra-libs.sh
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ user-mode-linux/arch/um/scripts/extra-libs.sh	2019-12-30 16:07:03.855845555 +0530
@@ -0,0 +1,10 @@
+#!/bin/sh
+
+# This file should be included from link-vmlinux, not executed!!!
+
+if [ "${CONFIG_UML_NET_VDE}" = "y" ] ; then
+       UML_EXTRA_LIBS="$UML_EXTRA_LIBS -lvdeplug"
+fi
+if [ "${CONFIG_UML_NET_PCAP}" = "y" ] ; then
+       UML_EXTRA_LIBS="$UML_EXTRA_LIBS -lpcap"
+fi
Index: user-mode-linux/scripts/link-vmlinux.sh
===================================================================
--- user-mode-linux.orig/scripts/link-vmlinux.sh	2019-12-30 16:07:03.859845658 +0530
+++ user-mode-linux/scripts/link-vmlinux.sh	2019-12-30 16:10:46.949229541 +0530
@@ -90,11 +90,13 @@
 			-Wl,--end-group				\
 			${@}"
 
+                . arch/um/scripts/extra-libs.sh
+
 		${CC} ${CFLAGS_vmlinux}				\
 			-o ${output}				\
 			-Wl,-T,${lds}				\
 			${objects}				\
-			-lutil -lrt -lpthread
+			-lutil -lrt -lpthread ${UML_EXTRA_LIBS}
 		rm -f linux
 	fi
 }
